clientapi/routing/memberships.go

```diff
@@ -57,10 +58,45 @@ func GetMemberships(
 	_ *config.ClientAPI,
 	rsAPI api.RoomserverInternalAPI,
 ) util.JSONResponse {
+	membership := req.URL.Query().Get("membership")
+	notMembership := req.URL.Query().Get("not_membership")
+
+	regexpMembershipFilter, _ := regexp.Compile("join|invite|leave|ban")
```

Comment:
This is insufficient for validation. Strings like `joinaaa` will pass. You need to use `^` and `$` to ensure there are no other characters in the string. In addition, this can be done statically and you can use `regexp.MustCompile` to avoid building the regexp on every request.